### PR TITLE
added logic to extract X-B3-TraceId from /put api header and put into MDC param

### PR DESCRIPTION
**Pull request by shabd67**
# Resolves

[CERES-640](https://rackspace.atlassian.net/browse/CERES-640)

# What

Distributed logging implementation in ceres and ingestor app

# How

added logic to extract X-B3-TraceId from /put api header and put into MDC param

### CODE REVIEW

**Review by rohitsngh27 — APPROVED** [review]
LGTM

File: src/main/java/com/rackspace/ceres/app/web/MDCFilter.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+@Component
+@Slf4j
+public class MDCFilter implements WebFilter {
+
+  private static final String X_B3_TRACE_ID = "X-B3-TraceId";
+
+  @Override
+  public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
+    MDC.put(X_B3_TRACE_ID, exchange.getRequest().getHeaders().getFirst(X_B3_TRACE_ID));
```
Comment (iWebi) [inline src/main/java/com/rackspace/ceres/app/web/MDCFilter.java:36]: Will this cause issues if header is not sent from clients? Trace Id header may or may not be included in the request. Isn't it?

**Review by iWebi — APPROVED** [review]
LGTM. Just a question about null value for trace header.

File: src/main/java/com/rackspace/ceres/app/web/MDCFilter.java (hunk 1 above)
Comment (shabd67) [reply src/main/java/com/rackspace/ceres/app/web/MDCFilter.java:36]: yes. if header is not present then the logs will be like:

2021-12-03 20:36:00.721 TRACE traceId=debc01960d097cb0, spanId=debc01960d097cb0, X-B3-TraceId=null 77434 --- [ctor-http-nio-5] c.r.ceres.app.services.DataWriteService  : Ingesting metric=Metric(timestamp=2021-12-01T13:15:12.050Z, metric=cpu_idle22, tags={host=h2, metricGroup=Misc, resource=test-resource}, value=189) for tenant=t-1

**Review by iWebi — APPROVED** [review]

**Pull request merged**